### PR TITLE
Implements another lighting optimization from bay.

### PR DESCRIPTION
**Pull request by PJB3005**
https://github.com/Baystation12/Baystation12/pull/10070

~~DNM IT'S BROKEN AAAAHA~~

### CODE REVIEW

File: code/modules/lighting/light_source.dm [hunk 1]
```diff
@@ -178,12 +184,15 @@
 /datum/light_source/proc/remove_lum()
 	applied = 0
 
+	var/i
```
Comment (RemieRichards) [inline code/modules/lighting/light_source.dm]: This needs to start at 1, otherwise you're doing ```-effect_r[null],-effect_g[null],-effect_b[null]```

Comment (d3athrow) [inline code/modules/lighting/light_source.dm]: jesus

Comment (PJB3005) [inline code/modules/lighting/light_source.dm]: that'll be why it didn't work

File: code/modules/lighting/light_source.dm [hunk 2]
```diff
@@ -169,6 +170,11 @@
 					round(lum_b * ., LIGHTING_ROUND_VALUE)
 				)
 
+			else
+				effect_r += 0
+				effect_b += 0
+				effect_g += 0
```
Comment (d3athrow) [inline code/modules/lighting/light_source.dm]: Is this real life, or a fanta sea?

Comment (RemieRichards) [inline code/modules/lighting/light_source.dm]: lol that's in the original Bay PR too.

Comment (PJB3005) [inline code/modules/lighting/light_source.dm]: what?

Comment (Intigracy) [inline code/modules/lighting/light_source.dm]: What's the point of adding 0?

Comment (RemieRichards) [inline code/modules/lighting/light_source.dm]: @PJB3005 I'll assume your maths is good enough to realise that ```a += 0``` is still ```a```

Comment (PJB3005) [inline code/modules/lighting/light_source.dm]: that's a list.

Comment (RemieRichards) [inline code/modules/lighting/light_source.dm]: Yes but that's a weird fucking syntax for adding a 0 as an element to a list really.

Comment (PJB3005) [inline code/modules/lighting/light_source.dm]: I don't see how that's weird

Comment (RemieRichards) [inline code/modules/lighting/light_source.dm]: to read the code you need to know that's a list.
```variable += object``` is OBVIOUSLY a list, as you cant store objects in anything else (storing an object in an object is just the contents list)
where as ```variable += integer``` at first glance will be read as a mathematical operation.

Comment (Exxion) [inline code/modules/lighting/light_source.dm]: Could use Add() instead of += for clarity

Comment (PJB3005) [inline code/modules/lighting/light_source.dm]: it takes 5 seconds to figure out that's a list, scroll up to the top and see effect_r is a list

**Pull request merged**